src/services/user.ts

```diff
@@ -106,14 +109,53 @@ export class UserService implements IUserService {
   }
 
   public async loginOrSignup(login: string, password: string): Promise<IUserPublic> {
-    const userDb = await User.findOne({ login }).exec();
+    const userDb = await this.findUserInDb(login);
 
     if (userDb) {
       return this.login(login, password);
     }
     return this.signup(login, password);
   }
 
+  public async requestResetPassword(login: string): Promise<IUserResetPassword> {
+    this.validateEmail(login);
+    const userDb = await this.findUserInDb(login);
+    if (userDb) {
+      const secret = JWT_SECRET + userDb.passwordHash;
+      const paylod = {
+        email: login,
+        id: userDb._id,
+      };
+      const token = jwt.sign(paylod, secret, { expiresIn: '1h' });
+      const resetUrl = '?token=' + token + '&email=' + login;
+      sendResetPasswordEmail(login, resetUrl);
```

Comment:
order matters. why are you sending the email in the middle of other operations? you should do it as the last operation before returning. what happens if there's an error in in some statement below? you will send an email but the process would fail.
also to separate operations that are not directly tied together I put empty lines and you should too. so: 
- construct data
- empty line
- send email
- empty line
- return